project-root/Assets/Scripts/NavMeshSpawner.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NavMeshSpawner : MonoBehaviour
+{
+    public GameObject monsterSpawn;
+    Vector3 point;
+    public float time;
+    public bool done;
+    public List<GameObject> monsters = new List<GameObject>();
+    //public int enemyIndex;
+    public int spawned;
+    public int amountToSpawn;
+    public int reSpawnTime;
+
+    public float range = 10f;
+
+    bool RandomPoint(Vector3 center, float range, out Vector3 result)

```

Comment:
Put any methods that are being declared after your Start() or Update() methods for readability. It's a common practice to do that so you should fix this and get used to it.